apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/InpatientClaimTransformerV2Test.java

```diff
@@ -71,15 +72,19 @@ public InpatientClaim generateClaim() throws FHIRException {
   @Before
   public void before() {
     claim = generateClaim();
-    eob = InpatientClaimTransformerV2.transform(new MetricRegistry(), claim, Optional.empty());
+    ExplanationOfBenefit genEob =
+        InpatientClaimTransformerV2.transform(new MetricRegistry(), claim, Optional.empty());
+    IParser parser = fhirContext.newJsonParser();
+    String json = parser.encodeResourceToString(genEob);
+    eob = parser.parseResource(ExplanationOfBenefit.class, json);
   }
 
   private static final FhirContext fhirContext = FhirContext.forR4();
 
   /** Common top level EOB values */
   @Test
   public void shouldSetID() {
-    Assert.assertEquals("inpatient-" + claim.getClaimId(), eob.getId());
+    Assert.assertEquals("inpatient-" + claim.getClaimId(), eob.getIdElement().getIdPart());
```

Comment:
This assertion is a little bit different from the other similar ones in other classes. Is that by design?

Reply:
Yes. Either should work. Just wanted to make sure one checked the actual ID object.